src/pytorch_lightning/callbacks/progress/base.py

```diff
@@ -172,6 +172,27 @@ def total_val_batches(self) -> Union[int, float]:
         assert self._trainer is not None
         return sum(self.trainer.num_val_batches) if self._trainer.fit_loop.epoch_loop._should_check_val_epoch() else 0
 
+    @property
+    def total_main_progress_bar_count_current_epoch(self) -> Union[int, float]:
+        total_train_batches = self.total_train_batches
+        total_val_batches = self.total_val_batches
+        assert self._trainer is not None
+
+        if total_train_batches != float("inf") and total_val_batches != float("inf"):
+            # val can be checked multiple times per epoch
+            val_check_batch = self.trainer.val_check_batch
+            if self.trainer.check_val_every_n_epoch is None:
+                batches_that_stepped = self.trainer.fit_loop.epoch_loop._batches_that_stepped
```

Comment:
Why do we use batches_that_stepped instead of the number of batches processed? Batches where we don't `optimizer.step` should also be counted?

Reply:
they are... `batches_that_stepped` doesn't care about `optimizer.step`.

```py
def training_step(...):
    return None

batches_that_stepped > 0
```

Reply:
But batches where we accumulate don't

https://github.com/Lightning-AI/lightning/blob/d78698528de9af9177dd7444677c97e1b0171515/src/pytorch_lightning/loops/epoch/training_epoch_loop.py#L247

Reply:
yep!

then we need to update the logic first
https://github.com/Lightning-AI/lightning/blob/d78698528de9af9177dd7444677c97e1b0171515/src/pytorch_lightning/loops/epoch/training_epoch_loop.py#L519-L522

do you agree?

Reply:
okay. yeah, it makes sense.. logic should be updated too

Reply:
done!